ssz_test.go

```diff
@@ -767,6 +767,22 @@ func toBytes32(x []byte) [32]byte {
 	return y
 }
 
+func TestBoolArrayMissingByte(t *testing.T) {
+	objBytes := hexDecodeOrDie(t, "010101010101010101010101010101")
+	var result [16]bool
+	if err := Unmarshal(objBytes, &result); err == nil {
+		t.Fatal("expected message with missing byte to fail marshalling")
+	}
+}
+
+func TestBoolArrayExtraByte(t *testing.T) {
+	objBytes := hexDecodeOrDie(t, "01010101010101010101010101010101ff")
+	var result [16]bool
+	if err := Unmarshal(objBytes, &result); err == nil {
+		t.Fatal("expected message with extra byte to fail marshalling")
+	}
+}
+
```

Comment:
maybe add a test for bool array that passes?

Reply:
Added